pkg/policy/resolve.go

```diff
@@ -193,13 +193,9 @@ func (p *EndpointPolicy) computeDirectionL4PolicyMapEntries(policyMapState MapSt
 	}
 }
 
-// ConsumeMapChanges transfers the changes from MapChanges to the caller,
-// locking the selector cache to make sure concurrent identity updates
-// have completed.
-// PolicyOwner (aka Endpoint) is also locked during this call.
+// ConsumeMapChanges transfers the changes from MapChanges to the caller.
+// PolicyOwner (aka Endpoint) is locked during this call.
 func (p *EndpointPolicy) ConsumeMapChanges() (adds, deletes Keys) {
-	p.selectorPolicy.SelectorCache.mutex.Lock()
-	defer p.selectorPolicy.SelectorCache.mutex.Unlock()
```

Comment:
I don't think this can be done. It will introduce race conditions with concurrent identity updates as the removed comments states. `ConsumeMapChanges` doesn't have an explicit dependency on the SelectorCache, but it's implicit. I can understand why someone might think that these statements can be removed.

Reply:
That was my doubt too. It wasn't clear to me whether it was still necessary after the rework, since there seemed to be no explicitly dependency. Thanks for the clarification